### ISSUE TITLE
API for manually setting name of an endpoint/job

### ISSUE THREAD

**Issue by cschneid** [scoutapp/scout_apm_ruby#140]
We've run into the use case of a central Controller or Job that then determines the actual work to do based on parameters (think `GraphQL` controller, or a `EventProcessor` background job). Since they're acting as user-driven routers, it's more useful to name the work done based on the result of that routing, not just the router itself.

Right now, we have a rough way to rename a transaction:

```ruby
req = ScoutApm::RequestManager.lookup
req.current_layer.name = "Override"
```

But that has limitations, and works only if the current_layer is exactly the `Controller` or `Job` layer.

A better api could be created, making it easier to use, and also hard to misuse (by checking the arity of that name and other similar gotchas).

**Comment by alindeman** [comment]
What is your sense of what the API should look like? A new method on the `TrackedRequest`, like it looks like [we've recommended before](https://github.com/scoutapp/scout_apm_ruby/blob/master/CHANGELOG.markdown#2114), perhaps?

**Comment by cschneid** [comment]
I think an API that works more like how `Context` works would be best.

```
ScoutApm::Transaction.ignore!
ScoutApm::Transaction.rename("User Lookup")
```

(the word Transaction since then it works for both web requests and background jobs).

Also note that `rename` should be smarter than simply changing current layer. It should go find the Controller or Job layer and rename that. That way we don't accidentally rename a different, nested, layer when called. Makes it less fidlly for users.

**Comment by cschneid** [comment]
Closed by #162
